test/Atmos/EDMF/compute_mse.jl

```diff
@@ -0,0 +1,221 @@
+using ClimateMachine
+const clima_dir = dirname(dirname(pathof(ClimateMachine)));
+
+plot_comparison = get(ARGS, 1, false)
+if plot_comparison
+    using Plots
+end
+
+using Test
+using NCDatasets
+using Dierckx
+using PrettyTables
+using Printf
+using ClimateMachine.ArtifactWrappers
+
+# Get PyCLES_output dataset folder:
+#! format: off
+PyCLES_output_dataset = ArtifactWrapper(
+    joinpath(clima_dir, "test", "Atmos", "EDMF", "Artifacts.toml"),
+    "PyCLES_output",
+    ArtifactFile[
+    # ArtifactFile(url = "https://caltech.box.com/shared/static/johlutwhohvr66wn38cdo7a6rluvz708.nc", filename = "Rico.nc",),
+    # ArtifactFile(url = "https://caltech.box.com/shared/static/zraeiftuzlgmykzhppqwrym2upqsiwyb.nc", filename = "Gabls.nc",),
+    # ArtifactFile(url = "https://caltech.box.com/shared/static/toyvhbwmow3nz5bfa145m5fmcb2qbfuz.nc", filename = "DYCOMS_RF01.nc",),
+    # ArtifactFile(url = "https://caltech.box.com/shared/static/ivo4751camlph6u3k68ftmb1dl4z7uox.nc", filename = "TRMM_LBA.nc",),
+    # ArtifactFile(url = "https://caltech.box.com/shared/static/4osqp0jpt4cny8fq2ukimgfnyi787vsy.nc", filename = "ARM_SGP.nc",),
+    ArtifactFile(url = "https://caltech.box.com/shared/static/jci8l11qetlioab4cxf5myr1r492prk6.nc", filename = "Bomex.nc",),
+    # ArtifactFile(url = "https://caltech.box.com/shared/static/pzuu6ii99by2s356ij69v5cb615200jq.nc", filename = "Soares.nc",),
+    # ArtifactFile(url = "https://caltech.box.com/shared/static/7upt639siyc2umon8gs6qsjiqavof5cq.nc", filename = "Nieuwstadt.nc",),
+    ],
+)
+PyCLES_output_dataset_path = get_data_folder(PyCLES_output_dataset)
+data_files = Dict()
+data_files[:Bomex] = Dataset(joinpath(PyCLES_output_dataset_path, "Bomex.nc"), "r")
+# data_files[:Rico] = Dataset(joinpath(PyCLES_output_dataset_path, "Rico.nc"), "r")
+# data_files[:Gabls] = Dataset(joinpath(PyCLES_output_dataset_path, "Gabls.nc"), "r")
+# data_files[:DYCOMS_RF01] = Dataset(joinpath(PyCLES_output_dataset_path, "DYCOMS_RF01.nc"), "r")
+# data_files[:TRMM_LBA] = Dataset(joinpath(PyCLES_output_dataset_path, "TRMM_LBA.nc"), "r")
+# data_files[:ARM_SGP] = Dataset(joinpath(PyCLES_output_dataset_path, "ARM_SGP.nc"), "r")
+# data_files[:Soares] = Dataset(joinpath(PyCLES_output_dataset_path, "Soares.nc"), "r")
+# data_files[:Nieuwstadt] = Dataset(joinpath(PyCLES_output_dataset_path, "Nieuwstadt.nc"), "r")
+#! format: on
+
+include("variable_map.jl")
+
+function compute_mse(
+    grid,
+    bl,
+    time_cm,
+    dons_arr,
+    ds,
+    experiment,
+    best_mse,
+    plot_dir = nothing,
+)
+    mse = Dict()
+
+    # Ensure domain matches:
+    z_les = ds["z_half"][:]
+    z_cm = get_z(grid; rm_dupes = true)
+    @test maximum(z_les) + last(diff(z_les)) / 2 ≈ z_cm[end]
+    time_les = ds["t"][:]
+
+    # Find the nearest matching final time:
+    t_min = min(time_cm[end], time_les[end])
+    i_t_cm = argmin(abs.(t_min .- time_cm))
+    i_t_les = argmin(abs.(t_min .- time_les))
+    t_cm = time_cm[i_t_cm]
+    t_les = time_les[i_t_les]
```

Comment:
The only situation I am concerned about is when we might get worse MSE because we change the CFL such that `abs(t_cm-t_les)` is larger than in a previous experiment. Then the MSE will grow but not because the solution is worse, just because we are testing two times that are farther in time.

Reply:
Agreed. I just realized that we're using the 2D spline, so we can interpolate the data to the same time for the comparison. Also, I'll add a test to ensure `t_compare > 400` so that we don't pass just by running a shorter simulation.